include/libdnf/system/state.hpp

```diff
@@ -41,7 +41,7 @@ class State {
     /// where the state is stored.
     /// @param dir_path The directory where the state is stored.
     /// @since 5.0
-    State(const std::filesystem::path & installroot, const std::filesystem::path & dir_path = "var/lib/dnf/state");
+    State(const std::filesystem::path & path);
```

Comment:
Why system_state is a public class or at least why it has a public constructor and methods for modifications and save? I gues it would be nice to use transactional approach where changes in system state will be recorded in history DB. What do you think?

Reply:
There's a TODO above the class definition. The API for retrieving the state should be public, so this will need some more work.

Reply:
Ok, TODO is fine.